src/wallet/rpcdump.cpp

```diff
@@ -1263,3 +1444,145 @@ UniValue importmulti(const JSONRPCRequest& mainRequest)
+            "\nResult (same as for importmulti):\n"
+            "[{ \"success\": true }"
+        );
+    }
+
+    ExternalSigner *signer = GetSignerForJSONRPCRequest(request, 0, pwallet);
+
+    UniValue descriptors = UniValue(UniValue::VARR);
+    std::string desc_prefix = "";
+    std::string desc_suffix = "";
+    std::string purpose = "";
+    switch(pwallet->m_default_address_type) {
+        case OutputType::LEGACY: {
+            desc_prefix = "pkh(";
+            desc_suffix = ")";
+            purpose = "44h";
+            break;
+        }
+        case OutputType::P2SH_SEGWIT: {
+            desc_prefix = "sh(wpkh(";
+            desc_suffix = "))";
+            purpose = "49h";
+            break;
+        }
+        case OutputType::BECH32: {
+            desc_prefix = "wpkh(";
+            desc_suffix = ")";
+            purpose = "84h";
+            break;
+        }
+        case OutputType::CHANGE_AUTO: {
+            assert(false);
+        }
+    }
+
+    const std::string receive_desc = desc_prefix + signer->m_fingerprint + "/" + purpose + "/" + (signer->m_mainnet ? "0h" : "1h") + "/0h/0/*" + desc_suffix;
+    UniValue receive_descriptors = signer->getKeys(receive_desc);
+    if (!receive_descriptors.isArray()) JSONRPCError(RPC_WALLET_ERROR, "Expected an array of receive descriptors");
+    for (const UniValue& descriptor : receive_descriptors.getValues()) {
+        descriptors.push_back(descriptor);
+    }
+
+
+    switch(pwallet->m_default_change_type) {
+        case OutputType::LEGACY: {
+            desc_prefix = "pkh(";
+            desc_suffix = ")";
+            purpose = "44h";
+            break;
+        }
+        case OutputType::P2SH_SEGWIT: {
+            desc_prefix = "sh(wpkh(";
+            desc_suffix = "))";
+            purpose = "49h";
+            break;
+        }
+        case OutputType::BECH32: {
+            desc_prefix = "wpkh(";
+            desc_suffix = ")";
+            purpose = "84h";
+            break;
+        }
+        case OutputType::CHANGE_AUTO: {
+            // Use same values as for receive descriptor
+            break;
+        }
+    }
+
+    const std::string change_desc = desc_prefix + signer->m_fingerprint + "/" + purpose + "/" + (signer->m_mainnet ? "0h" : "1h") + "/0h/1/*" + desc_suffix;
+    UniValue change_descriptors = signer->getKeys(change_desc);
+    if (!change_descriptors.isArray()) JSONRPCError(RPC_WALLET_ERROR, "Expected an array of change descriptors");
+    for (const UniValue& descriptor : change_descriptors.getValues()) {
+        descriptors.push_back(descriptor);
+    }
+
+    if (receive_descriptors.size() != change_descriptors.size()) JSONRPCError(RPC_WALLET_ERROR, "Expected same number of receive and change descriptors");
+
+    // Use importmulti to process the descriptors:
+    UniValue importdata(UniValue::VARR);
+
+    uint64_t keypool_target_size = 0;
+    keypool_target_size = gArgs.GetArg("-keypool", DEFAULT_KEYPOOL_SIZE);
```

Comment:
Make signedness changing implicit conversion explicit?

Reply:
The compiler didn't like it when I initialized using `gArgs.GetArg`.